src/k8s/pkg/k8sd/api/cluster_config.go

```diff
@@ -0,0 +1,29 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+
+	apiv1 "github.com/canonical/k8s/api/v1"
+	"github.com/canonical/lxd/lxd/response"
+	"github.com/canonical/microcluster/rest"
+	"github.com/canonical/microcluster/state"
+)
+
+var k8sdClusterConfig = rest.Endpoint{
+	Path: "k8sd/config",
+	Get:  rest.EndpointAction{Handler: clusterConfigGet, AllowUntrusted: false},
+}
+
+func clusterConfigGet(s *state.State, r *http.Request) response.Response {
+	config, err := os.ReadFile("/etc/kubernetes/admin.conf")
```

Comment:
should we instead render a new kubeconfig?

Reply:
not something for now, we could simply update the implementation later

Reply:
Good point, I added a TODO